File: app/src/main/java/dev/exit/asset/config/CurrentUserProvider.java
package dev.exit.asset.config;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
import org.springframework.stereotype.Component;

@Component
public class CurrentUserProvider {

	public String ownerId() {
		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
		if (authentication instanceof JwtAuthenticationToken jwtAuthentication) {
			return jwtAuthentication.getToken().getSubject();
		}
		return "demo-user";
	}
}

File: common/src/main/kotlin/dev/techwrite/platform/common/CurrentUserProvider.kt
package dev.techwrite.platform.common

import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
import org.springframework.stereotype.Component

@Component
class CurrentUserProvider {

    fun ownerId(): String {
        val authentication = SecurityContextHolder.getContext().authentication
        if (authentication is JwtAuthenticationToken) {
            return authentication.token.subject
        }
        return "demo-user"
    }
}
